### ISSUE TITLE
unnecessary counters are added to the var names inside local functions

### ISSUE THREAD

**Issue by nadako** [HaxeFoundation/haxe#7237]
this is minor and probably was there for a long time, but we might want to fix this someday:

```haxe
class Test {
  static function main() {
    var f = function(a) {
      var b = a;
      return b;
    }
    var f2 = function(a) {
      var b = a;
      return b;
    }
  }
}
```

generates:

```js
Test.main = function() {
	var f = function(a) {
		var b = a;
		return b;
	};
	var f2 = function(a1) {
		var b1 = a1;
		return b1;
	};
};
```

it can get a bit ridiculous when macro-generating a lot of code, e.g.:
```js
description1346.addInjectionMethod(function(target1083,injector1063) {
	target1083.campaignModel = injector1063.getInstanceForMapping("de.innogames.strategycity.campaignmap.model.CampaignModel");
	target1083.userDataModel = injector1063.getInstanceForMapping("de.innogames.strategycity.main.model.UserDataModel");
});
```

**Comment by nadako** [comment]
I'll raise the priority here a bit because it would make our life at @innogames a bit easier :) And it seems that #9045 will fix this anyway \o/

**Comment by nadako** [comment]
The issue is fixed by #9045, but we probably want to add a test for this so it doesn't regress.
